nmigen_boards/de0_nano.py

```diff
@@ -0,0 +1,73 @@
+import os
+import subprocess
+
+from nmigen.build import *
+from nmigen.vendor.intel import *
+from .resources import *
+
+__all__ = ["DE0NanoPlatform"]
+
+
+class DE0NanoPlatform(IntelPlatform):
+    device      = "EP4CE22" # Cyclone IV 22K LEs
+    package     = "F17"   # FBGA-256
+    speed       = "C6"
+    default_clk = "clk50"
+    resources   = [
+        Resource("clk50", 0, Pins("R8", dir="i"),
+                 Clock(50e6), Attrs(io_standard="3.3-V LVTTL")),
+
+        Resource("i2c_flash", 0,
+            Subsignal("clk", Pins("F2", dir="o")),
+            Subsignal("dat", Pins("F1", dir="io")),
+            Attrs(io_standard="3.3-V LVTTL")
+        ),
+
+        Resource("ADC", 0,
```

Comment:
Could you link to the datasheet of the ADC please?

Reply:
https://www.ti.com/lit/ds/symlink/adc128s022.pdf looks like it's just an SPI interface. Should this be using SPIResources instead?

Reply:
modified to use an SPIResource using the ULX3S board file as an example